### PR TITLE
Systematic Uncertainty Update

### PR DESCRIPTION
**Pull request by DouglasBurns**
Remove the options for separate lists of systematics simultaneously. Can just create them and run one after the other anyway...
add the renormalisation and factorisation scale variations in
Create an envelope of the scale variations
Make PDF uncertainty from the RMS of the variations.
Realised that we were getting the symmetrised uncertainty by finding the max difference between the central value and the pdf uncertainty, not the central value and upper PDF variation... Rearranged so that PDF uncertainty is now calculated in the correct place.
Correctly find PU variation files
Correctly find LightJet variation files
Combine BJet and Light Jet syst uncertainties

### CODE REVIEW

**Comment by DouglasBurns** [comment]
Ready for review

File: tools/systematic.py [hunk 1]
```diff
@@ -116,107 +125,221 @@ def get_systematic_measured_values_only(options, syst_x_secs_and_errs):
...
+
+    # Separate into up/down scale variations
+    for scale_variation in d_syst:
+        if 'down' in scale_variation:
+            down_variations.append(d_syst[scale_variation])
+        elif 'up' in scale_variation:
+            up_variations.append(d_syst[scale_variation])
+
+    # find min/max
+    for v1, v2, v3 in zip (up_variations[0], up_variations[1], up_variations[2]):
```
Comment (kreczko) [inline tools/systematic.py]: I fail to understand why where are always 3 entries in the up_variation list. From the code above I would then to be `len(d_syst)` of them.

Can you maybe explain what `d_syst` is in the doc for the function?

File: tools/systematic.py [hunk 2]
```diff
@@ -255,6 +382,7 @@ def scaleTopMassSystematic( upper_uncertainty, lower_uncertainty, topMasses, top
 
 def get_sign(central, upper, lower, upper_variation, lower_variation):
     '''
+    Currently Obsolete.
```
Comment (kreczko) [inline tools/systematic.py]: can you add a `@deprecated` decorator? This way python will produce a warning if the function is used somewhere.

File: tools/systematic.py [hunk 3]
```diff
@@ -266,28 +394,33 @@ def get_sign(central, upper, lower, upper_variation, lower_variation):
...
+        for systematic, measurement in x_sec_with_symmetrised_systematics.iteritems():
+            if (systematic == 'central'): continue
+            sys_unc += measurement[0][bin_i]**2
+        measurement_with_total_uncertainty.append( [central[0], sqrt(sys_unc), sqrt(sys_unc)] )
     return measurement_with_total_uncertainty
 
 def print_dictionary(title, dictionary_to_print):
     '''
     Prints dictionaries in a nicer form
+    TODO Maybe think how pandas can be incorporated into 03...
```
Comment (kreczko) [inline tools/systematic.py]: In a way. As you can see we are passing too many lists around.
I would be much happier if we can make them in to more complex objects that are easier to understand (e.g. a table).

File: tools/systematic.py (hunk 1 above)
Comment (DouglasBurns) [inline tools/systematic.py]: So the d_syst is a dictionary containing all Q2 scale variations (muR up/down, muF up/down muRmuF up/down) of which all the up(down) variations are stored in up(down)_variations[]. As these are the only variations passed to the function the up(down)_variations will always be length 3. I will update the doc to reflect this

File: config/cross_section_config.py [hunk 4]
```diff
@@ -415,6 +425,14 @@ def __fill_defaults__( self ):
         self.unfolding_matching_down = self.unfolding_matching_down_raw.replace( '.root', '_asymmetric.root' )
         self.unfolding_matching_up = self.unfolding_matching_up_raw.replace( '.root', '_asymmetric.root' )
 
+        self.unfolding_renormalisation_down = path_to_unfolding_histograms + 'unfolding_TTJets_%dTeV_asymmetric_05muR1muF.root' % self.centre_of_mass_energy
```
Comment (DouglasBurns) [inline config/cross_section_config.py]: OK i will look into after this PR

File: tools/systematic.py (hunk 3 above)
Comment (DouglasBurns) [inline tools/systematic.py]: This is a little more difficult in pandas due to the amount of nesting occuring in the dictionaries/lists. It is something I am happy to look into but i think also deserves a separate PR

File: tools/systematic.py [hunk 5]
```diff
@@ -382,6 +385,7 @@ def scaleTopMassSystematic( upper_uncertainty, lower_uncertainty, topMasses, top
 
 def get_sign(central, upper, lower, upper_variation, lower_variation):
     '''
+    @deprecated
```
Comment (kreczko) [inline tools/systematic.py]: That is not how you apply a decorator ;).
The decorator goes directly above `dev get_sign`:
http://www.learnpython.org/en/Decorators
